src/library/crate/cratestorage.cpp

```diff
@@ -0,0 +1,665 @@
+#include "library/crate/cratestorage.h"
+
+#include "library/crate/crateschema.h"
+#include "library/dao/trackschema.h"
+
+#include "util/db/dbconnection.h"
+#include "util/db/sqltransaction.h"
+#include "util/db/fwdsqlquery.h"
+
+#include <QtDebug>
+
+
+namespace {
+
+const QString CRATETABLE_NAME = "name";
+const QString CRATETABLE_LOCKED = "locked";
+
+const QString CRATE_SUMMARY_VIEW = "crate_summary";
+
+const QString CRATESUMMARY_TRACK_COUNT = "track_count";
+const QString CRATESUMMARY_TRACK_DURATION = "track_duration";
+
+const QString kCrateTracksJoin = QString(
+        "LEFT JOIN %3 ON %3.%4=%1.%2").arg(
```

Comment:
It is a pity that we cannot build the strings at compile time.
What would be a solution using constexpr()?